tpm/ak.go

```diff
@@ -0,0 +1,204 @@
+package tpm
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"time"
+
+	"github.com/google/go-attestation/attest"
+
+	"go.step.sm/crypto/tpm/storage"
+)
+
+type AK struct {
```

Comment:
There's lots of places where you `return result, err`. Maybe turn result into a pointer and start returning nil?